File: Sorting/Javascript/countingSort.js
function countingSort(arr) {
    let size = arr.length;
    let numMax = arr[0];
    let numMin = arr[0];
    for (let i = 1; i < size; i++) {
        if (arr[i] > numMax)
            numMax = arr[i];
        if (arr[i] < numMin)
            numMin = arr[i];
    }
    let sizeTemp = numMax - numMin + 1;
    let tempArr = new Array(sizeTemp);
    for (var i = 0; i < sizeTemp; i++)
        tempArr[i] = 0;
    for (var i = 0; i < size; i++)
        tempArr[arr[i] - numMin]++;
    let counter = 0;
    for (var i = 0; i < sizeTemp; i++)
        for (var j = 0; j < tempArr[i]; j++) {
            arr[counter] = i + numMin;
            counter++;
        }
}

File: Sorting/Java/CountingSort.java
import java.util.*;

class CountingSort
{
    public static void main(String [] args)
    {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for(int i=0;i<n;i++)
            arr[i] = sc.nextInt();
        sort(arr);
        for(int i=0;i<n;i++)
            System.out.print(arr[i] + " ");
        System.out.print("\n");
    }
    public static void sort(int[] arr)
    {
        int min = arr[0], max = arr[0];
        int n = arr.length;
        for(int i=1;i<n;i++)
        {
            if(min > arr[i])
                min = arr[i];
            if(max < arr[i])
                max = arr[i];
        }
        int[] tempArr = new int[max-min+1];
        for(int i=0;i<max-min+1;i++)
            tempArr[i] = 0;
        for(int i=0;i<n;i++)
            tempArr[arr[i]-min]++;
        int c = 0;
        for(int i=0;i<=max-min;i++)
            for(int j=0;j<tempArr[i];j++)
            {
                arr[c] = i+min;
                c++;
            }
    }
}
